### ISSUE TITLE
Avoid requesting images for hidden takeovers

### ISSUE THREAD

**Issue by nottrobin** [canonical-websites/www.ubuntu.com#4143]
Currently, visitors to the www.ubuntu.com homepage are downloading the images for all the takeovers, even though the majority of them are hidden:

![image](https://user-images.githubusercontent.com/519935/46601462-0bdaa480-cae5-11e8-89b4-e265bc7705ff.png)

We should make sure images from takeovers that won't be displayed aren't downloaded.

**Comment by inuwa** [comment]
@nottrobin I have had a play around, we have to hacks. One is to set the takeover image to a background image  but it requires having two images for varying screen sizes. Also the other option is hacky (i.e. using javascript to inject image src onload) and we will only be saving 50kb. Is it worth it?

**Comment by nottrobin** [comment]
@inuwa it's not just the number of kilobytes, it's also the delay associated with making extra HTTP requests, which is often [more significant](https://stackoverflow.com/questions/9855545/http-requests-vs-file-size) than filesize. If you wanted to really make the case for it being insignificant, you could try doing some benchmarking and see how much difference it makes to real-world page load times - although these will still vary hugely for different users.

I don't consider using JavaScript to set the `src` to be really a hack, when we're already [using JavaScript](https://github.com/canonical-websites/www.ubuntu.com/blob/master/templates/base_index.html#L142) to hide the takeovers. It would just be 2-ish extra lines in that JS block.

But if @anthonydillon considers this to be more effort than it's worth for the performance gain, given he cares about performance possibly more than I do, then fair enough.

**Comment by inuwa** [comment]
@nottrobin thank you for the reply, there does not seem to be any performance hit on chrome though for the takeover images because they are svg files. I took a screenshot of network download times and download time are minute. 
![image](https://user-images.githubusercontent.com/43535482/48146625-53537c80-e2ad-11e8-9685-e12590495ec7.png)

If you wish for us to continue, I'll liaise with @anthonydillon in the morning. Cheers

**Comment by nottrobin** [comment]
@inuwa all the items you show there are loaded from cache, which is a pointless test. Yes our assets server does a good job of asking browsers to cache images, but most visitors to the homepage will be first-time visitors and so not have the images cached.

Here are real download times, for me, for some of those images:

![image](https://user-images.githubusercontent.com/519935/48197371-7c2b4e80-e34d-11e8-82b9-94041f161869.png)

And here is a breakdown the top image - one of the smaller images at 5.2kb:

![image](https://user-images.githubusercontent.com/519935/48197453-ae3cb080-e34d-11e8-9016-1a7656df8778.png)

You can see my point illustrated here - the actual download only took 21ms, but the connection to perform the download spent 10ms "stalled" and 43ms "waiting" ([time to first byte](https://en.wikipedia.org/wiki/Time_to_first_byte)), so this incredibly tiny image actually tied up one of the browsers download processes for 75ms. Chrome is fairly good at concurrency, but in this case these images delayed the loading of the page by at least 100ms.

Bear in mind that I'm on a good >50mb connection, and I'm just a few miles from London (where our servers are). Now imagine you're in Nigeria, maybe 60 times further from London than I am, on a <1mb connection, using an older browser. Now those images could easily delay the load time of our homepage by more than a second, and might delay it by significantly more than that.

**Comment by anthonydillon** [comment]
Good point Robin. Let's try the JS solution for now. The only downfall is
we need to remember to keep it up.

I suggest leaving the image src blank. Adding the image URL to a data
attribute. Use JS to replace the src attr when revealing the takeover.

**Comment by nottrobin** [comment]
@anthonydillon one way to avoid the need to "keep it up" is to wrap all takeover markup in a [`<template>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template). Then the content won't be loaded until explicitly [imported](https://developer.mozilla.org/en-US/docs/Web/API/Document/importNode) with JavaScript.

I'd be happy to throw together this solution if you like.

This [won't work](https://caniuse.com/#feat=template) in Internet Explorer, which [is 3.24% of users over the last 3 months](https://analytics.google.com/analytics/web/#/report/content-pages/a1018242w108544136p113106543/_u.date00=20180810&_u.date01=20181108&explorer-table.plotKeys=%5B%5D&explorer-table.secSegmentId=analytics.browser&_r.drilldown=analytics.pagePath:www.ubuntu.com~2F/) but I think improving the performance for 96.76% of users is enough of a win here.

**Comment by anthonydillon** [comment]
@nottrobin Sure if you have time. Would be good to see it.
